src/main/java/report_use_cases/ReportInteractor.java

```diff
@@ -0,0 +1,59 @@
+package report_use_cases;
+
+import entities.Report;
+import entities.ReportFactory;
+import entities.Review;
+import entities.User;
+import report_screens.ManipulateReviewDs;
+import report_screens.ReportCreationFailure;
+import java.time.LocalDateTime;
+
+public class ReportInteractor implements reportInputBoundary {
+
+    final reportDsGateway reportDsGateway;
+
+    final ReportFactory reportFactory;
+
+    final Excalibur excalibur;
+
+
+    public ReportInteractor(reportDsGateway reportDsGateway, ReportFactory reportFactory, Excalibur excalibur) {
+        this.reportDsGateway = reportDsGateway;
+        this.excalibur = excalibur;
+        this.reportFactory = reportFactory;
+    }
+
+    @Override
+    public void Create(ReportRequestModel reportRequestModel){
+
+        //if report already exists
+        if(reportDsGateway.existsReportByReporterAndReview(reportRequestModel.getReporter().getUsername(), reportRequestModel.getReview().getID())){
+            throw new ReportCreationFailure("Report failed: You've already reported this review");
```

Comment:
Should these be exceptions? I don't think this counts as exceptional circumstances

Reply:
I think these should be. They are runtime exceptions; they will be raised when the user violates the requirements of sending a report (A user cannot be banned and identical report should not be sent again). The ReportScreen will catch those exceptions and show the error message (in terms of which specific violatoin is being made) to the user.